library/core/src/panic.rs

```diff
@@ -58,6 +58,42 @@ pub macro panic_2021 {
     ),
 }
 
+#[doc(hidden)]
+#[unstable(feature = "edition_panic", issue = "none", reason = "use unreachable!() instead")]
+#[allow_internal_unstable(core_panic, const_format_args)]
+#[rustc_diagnostic_item = "unreachable_2015_macro"]
+#[rustc_macro_transparency = "semitransparent"]
+pub macro unreachable_2015 {
+    () => (
+        $crate::panicking::panic("internal error: entered unreachable code")
+    ),
+    // Use `panic_str` instead of `panic_display::<&str>` for non_fmt_panic lint.
+    ($msg:expr $(,)?) => (
+        $crate::panicking::panic_str(
+            $crate::concat!("internal error: entered unreachable code: ", $msg))
+    ),
```

Comment:
Using `concat` rather than formatting with Display breaks code like this:

```rust
    let a = "hello";
    unreachable!(a);
```

Reply:
Fixed with the introduction of `unreachable_display!` and test added https://github.com/rust-lang/rust/pull/93179/commits/9264aa58d2cc38c6a497f19ef616b38c52b3fff6